In [1]:
region_mapping = {
    # Города федерального значения
    "г. Москва": "г. Москва",
    "Город Москва столица Российской Федерации город федерального значения": "г. Москва",
    "Москва в старых границах": "г. Москва",
    "г. Санкт-Петербург": "г. Санкт-Петербург",
    "Город Санкт-Петербург город федерального значения": "г. Санкт-Петербург",
    
    # Республики
    "Республика Адыгея": "Республика Адыгея (Адыгея)",
    "Республика Адыгея (Адыгея)": "Республика Адыгея (Адыгея)",
    "Республика Алтай": "Республика Алтай",
    "Республика Башкортостан": "Республика Башкортостан",
    "Республика Бурятия": "Республика Бурятия",
    "Республика Дагестан": "Республика Дагестан",
    "Республика Ингушетия": "Республика Ингушетия",
    "Кабардино-Балкарская Республика": "Кабардино-Балкарская Республика",
    "Республика Калмыкия": "Республика Калмыкия",
    "Карачаево-Черкесская Республика": "Карачаево-Черкесская Республика",
    "Республика Карелия": "Республика Карелия",
    "Республика Коми": "Республика Коми",
    "Республика Марий Эл": "Республика Марий Эл",
    "Республика Мордовия": "Республика Мордовия",
    "Республика Саха (Якутия)": "Республика Саха (Якутия)",
    "Республика Северная Осетия – Алания": "Республика Северная Осетия — Алания",
    "Республика Северная Осетия-Алания": "Республика Северная Осетия — Алания",
    "Республика Северная Осетия — Алания": "Республика Северная Осетия — Алания",
    "Республика Татарстан": "Республика Татарстан (Татарстан)",
    "Республика Татарстан (Татарстан)": "Республика Татарстан (Татарстан)",
    "Республика Тыва": "Республика Тыва",
    "Удмуртская Республика": "Удмуртская Республика",
    "Республика Хакасия": "Республика Хакасия",
    "Чеченская Республика": "Чеченская Республика",
    "Чувашская Республика": "Чувашская Республика — Чувашия",
    "Чувашская Республика - Чувашия": "Чувашская Республика — Чувашия",
    "Чувашская Республика — Чувашия": "Чувашская Республика — Чувашия",
    
    # Края
    "Алтайский край": "Алтайский край",
    "Агинский Бурятский округ (Забайкальский край)	":"Забайкальский край",
    "Забайкальский край": "Забайкальский край",
    "Камчатский край": "Камчатский край",
    "Краснодарский край": "Краснодарский край",
    "Красноярский край": "Красноярский край",
    "Пермский край": "Пермский край",
    "Приморский край": "Приморский край",
    "Ставропольский край": "Ставропольский край",
    "Хабаровский край": "Хабаровский край",
    
    # Области
    "Амурская область": "Амурская область",
    "Архангельская область": "Архангельская область",
    "Архангельская область без автономного округа": "Архангельская область",
    "Архангельская область (кроме Ненецкого автономного округа)": "Архангельская область",
    "Архангельская область (без АО)": "Архангельская область",
    "Астраханская область": "Астраханская область",
    "Белгородская область": "Белгородская область",
    "Брянская область": "Брянская область",
    "Владимирская область": "Владимирская область",
    "Волгоградская область": "Волгоградская область",
    "Вологодская область": "Вологодская область",
    "Воронежская область": "Воронежская область",
    "Ивановская область": "Ивановская область",
    "Иркутская область": "Иркутская область",
    "Калининградская область": "Калининградская область",
    "Калинингpадская область": "Калининградская область",
    "Калужская область": "Калужская область",
    "Кемеровская область": "Кемеровская область — Кузбасс",
    "Кемеровская область - Кузбасс": "Кемеровская область — Кузбасс",
    "Кемеровская область — Кузбасс": "Кемеровская область — Кузбасс",
    "Кировская область": "Кировская область",
    "Костромская область": "Костромская область",
    "Курганская область": "Курганская область",
    "Курская область": "Курская область",
    "Ленинградская область": "Ленинградская область",
    "Липецкая область": "Липецкая область",
    "Магаданская область": "Магаданская область",
    "Московская область": "Московская область",
    "Московская обл. в старых границах": "Московская область",
    "Мурманская область": "Мурманская область",
    "Нижегородская область": "Нижегородская область",
    "Новгородская область": "Новгородская область",
    "Новосибирская область": "Новосибирская область",
    "Омская область": "Омская область",
    "Оренбургская область": "Оренбургская область",
    "Орловская область": "Орловская область",
    "Пензенская область": "Пензенская область",
    "Псковская область": "Псковская область",
    "Ростовская область": "Ростовская область",
    "Рязанская область": "Рязанская область",
    "Самарская область": "Самарская область",
    "Саратовская область": "Саратовская область",
    "Сахалинская область": "Сахалинская область",
    "Свердловская область": "Свердловская область",
    "Смоленская область": "Смоленская область",
    "Тамбовская область": "Тамбовская область",
    "Тверская область": "Тверская область",
    "Томская область": "Томская область",
    "Тульская область": "Тульская область",
    "Тюменская область": "Тюменская область",
    "Тюменская область   без автономных округов": "Тюменская область",
    "Тюменская область без автономных округов": "Тюменская область",
    "Тюменская область (без АО)": "Тюменская область",
    "Тюменская область (кроме Ханты-Мансийского автономного округа-Югры и Ямало-Ненецкого автономного округа)": "Тюменская область",
    "Ульяновская область": "Ульяновская область",
    "Челябинская область": "Челябинская область",
    "Ярославская область": "Ярославская область",
    
    # Автономные округа и область
    "Еврейская автономная область": "Еврейская автономная область",
    "Ненецкий автономный округ": "Ненецкий автономный округ",
    "Ненецкий автономный округ (Архангельская область)": "Ненецкий автономный округ",
    "Ханты-Мансийский автономный округ — Югра": "Ханты-Мансийский автономный округ — Югра",
    "Ханты-Мансийский     автономный округ – Югра": "Ханты-Мансийский автономный округ — Югра",
    "Ханты-Мансийский автономный округ – Югра": "Ханты-Мансийский автономный округ — Югра",
    "Ханты-Мансийский автономный округ - Югра (Тюменская область)": "Ханты-Мансийский автономный округ — Югра",
    "Чукотский автономный округ": "Чукотский автономный округ",
    "Ямало-Ненецкий автономный округ": "Ямало-Ненецкий автономный округ",
    "Ямало-Ненецкий автономный округ (Тюменская область)": "Ямало-Ненецкий автономный округ",
}

In [2]:
valid_regions = [
    "Республика Адыгея (Адыгея)", "Республика Алтай", "Республика Башкортостан",
    "Республика Бурятия", "Республика Дагестан", "Республика Ингушетия",
    "Кабардино-Балкарская Республика", "Республика Калмыкия", "Карачаево-Черкесская Республика",
    "Республика Карелия", "Республика Коми", "Республика Марий Эл",
    "Республика Мордовия", "Республика Саха (Якутия)", "Республика Северная Осетия — Алания",
    "Республика Татарстан (Татарстан)", "Республика Тыва", "Удмуртская Республика",
    "Республика Хакасия", "Чеченская Республика", "Чувашская Республика — Чувашия",
    "Алтайский край", "Забайкальский край", "Камчатский край", "Краснодарский край",
    "Красноярский край", "Пермский край", "Приморский край", "Ставропольский край",
    "Хабаровский край", "Амурская область", "Архангельская область", "Астраханская область",
    "Белгородская область", "Брянская область", "Владимирская область", "Волгоградская область",
    "Вологодская область", "Воронежская область", "Ивановская область", "Иркутская область",
    "Калининградская область", "Калужская область", "Кемеровская область — Кузбасс",
    "Кировская область", "Костромская область", "Курганская область", "Курская область",
    "Ленинградская область", "Липецкая область", "Магаданская область", "Московская область",
    "Мурманская область", "Нижегородская область", "Новгородская область", "Новосибирская область",
    "Омская область", "Оренбургская область", "Орловская область", "Пензенская область",
    "Псковская область", "Ростовская область", "Рязанская область", "Самарская область",
    "Саратовская область", "Сахалинская область", "Свердловская область", "Смоленская область",
    "Тамбовская область", "Тверская область", "Томская область", "Тульская область",
    "Тюменская область", "Ульяновская область", "Челябинская область", "Ярославская область",
    "г. Москва", "г. Санкт-Петербург", "Еврейская автономная область",
    "Ненецкий автономный округ", "Ханты-Мансийский автономный округ — Югра",
    "Чукотский автономный округ", "Ямало-Ненецкий автономный округ"
]

# Столбцы

## Объем сточных вод

In [3]:
import pandas as pd
df = pd.read_excel('Pril-Region_Pokaz_2024/Раздел 7 - Зем ресурсы и охр окр прир среды.xlsx', sheet_name=9, skiprows=5, skipfooter=3)
df.rename(columns={'Unnamed: 0':'Регион', '20142)':'2014'}, inplace=True)
df['Регион'] = df['Регион'].apply(lambda x: x.strip())
df.head()

Регион     2000     2001     2002     2003     2004  \
0           Российская Федерация  20291.0  19773.0  19767.0  18961.0  18534.0   
1  Центральный федеральный округ   4742.0   4692.0   5176.0   4478.0   4470.0   
2           Белгородская область     45.0     13.0      6.0     10.0     11.0   
3               Брянская область    100.0     99.0     96.0     94.0     92.0   
4           Владимирская область    182.0    182.0    185.0    174.0    169.0   

      2005     2006     2007     2008  ...      2014      2015      2016  \
0  17727.0  17489.0  17176.0  17119.0  ...  14697.77  14443.71  14744.16   
1   4341.0   4186.0   3870.0   3798.0  ...   3328.34   3227.25   3211.62   
2     11.0     11.0      8.0     11.0  ...     66.83     61.46     73.64   
3     89.0     96.0     94.0     85.0  ...     61.56     57.60     57.92   
4    155.0    147.0    141.0    140.0  ...    110.49    108.55    108.81   

       2017      2018      2019      2020      2021      2022     2023  
0  13615.54  13162.10  12626.62  11796.78  11613.83  11340.93  10842.0  
1   3168.25   3056.91   2903.28   2834.41   2864.58   2677.18   2735.0  
2     70.54     70.46     68.84     65.90     60.56     58.60     58.0  
3     55.12     53.78     50.75     49.69     51.40     52.06     53.0  
4    104.52     98.55     94.42     95.02     94.26     91.32     90.0  

[5 rows x 25 columns]

In [4]:
extra_regions = ['Российская Федерация', 'Архангельская область', 'Тюменская область', 'в том числе:', 'Республика Крым', 'г. Севастополь']
df = df[df['Регион'].str.endswith('федеральный округ') == False]
df = df[~df['Регион'].isin(extra_regions)] #~отрицание условия
df['Регион'] = df['Регион'].map(region_mapping)
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 83 entries, 2 to 97
Data columns (total 25 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2000    83 non-null     float64
 2   2001    83 non-null     float64
 3   2002    83 non-null     float64
 4   2003    83 non-null     float64
 5   2004    83 non-null     float64
 6   2005    83 non-null     float64
 7   2006    83 non-null     float64
 8   2007    83 non-null     float64
 9   2008    83 non-null     float64
 10  2009    83 non-null     float64
 11  2010    83 non-null     float64
 12  2011    83 non-null     float64
 13  2012    83 non-null     float64
 14  2013    83 non-null     float64
 15  2014    83 non-null     float64
 16  2015    83 non-null     float64
 17  2016    83 non-null     float64
 18  2017    83 non-null     float64
 19  2018    83 non-null     float64
 20  2019    83 non-null     float64
 21  2020    83 non-null     float64
 22  2021    8

## Инвестиции

Инвестиции 2000-2008

In [5]:
data1 = pd.read_excel("data (1).xls", skiprows=2)
data1.drop(0, inplace=True)
data1.rename(columns={'Unnamed: 0':'Регион', 'Unnamed: 1':'Вид оос'}, inplace=True)
data1['Регион'] = data1['Регион'].apply(lambda x: x.strip())
data1.drop(data1.iloc[:, 2:11].columns, axis=1, inplace=True)
data1 = pd.DataFrame(data1.drop('Вид оос', axis=1).groupby(['Регион']).sum())
data1.reset_index(inplace=True)
data1[data1.drop('Регион', axis=1).columns] = data1[data1.drop('Регион', axis=1).columns].astype(float)
extra_regions = ['Российская Федерация']
data1 = data1[data1['Регион'].str.endswith('федеральный округ') == False]
data1 = data1[~data1['Регион'].isin(extra_regions)] #~отрицание условия
data1 = data1[data1['Регион'].isin(region_mapping.keys())]
data1.loc[:, 'Регион'] = data1['Регион'].map(region_mapping)
data1.info()

<class 'pandas.core.frame.DataFrame'>
Index: 83 entries, 1 to 105
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2000    83 non-null     float64
 2   2001    83 non-null     float64
 3   2002    83 non-null     float64
 4   2003    83 non-null     float64
 5   2004    83 non-null     float64
 6   2005    83 non-null     float64
 7   2006    83 non-null     float64
 8   2007    83 non-null     float64
 9   2008    83 non-null     float64
dtypes: float64(9), object(1)
memory usage: 7.1+ KB


Инвестиции 2009-2016

In [6]:
data2 = pd.read_excel("data (2).xls", skiprows=2)
data2.drop(0, inplace=True)
data2.drop(['Unnamed: 0', 'Unnamed: 2'], axis=1, inplace=True)
data2.rename(columns={'Unnamed: 1':'Регион', 'Unnamed: 3':'Вид оос'}, inplace=True)
data2['Регион'] = data2['Регион'].apply(lambda x: x.strip())
data2 = pd.DataFrame(data2.drop('Вид оос', axis=1).groupby(['Регион']).sum())
data2.reset_index(inplace=True)
data2[data2.drop('Регион', axis=1).columns] = data2[data2.drop('Регион', axis=1).columns].astype(float)
data2

Регион         2009  \
0                                       Алтайский край    307775.00   
1                                     Амурская область   1269189.00   
2                                Архангельская область   1781604.10   
3    Архангельская область (кроме Ненецкого автоном...         0.00   
4                                 Астраханская область   2117091.00   
..                                                 ...          ...   
96               Южный федеральный округ (по 2009 год)  10266997.50   
97               Южный федеральный округ (с 2010 года)         0.00   
98              Южный федеральный округ (с 29.07.2016)         0.00   
99   Ямало-Ненецкий автономный округ (Тюменская обл...   8583498.94   
100                                Ярославская область   1987824.00   

            2010         2011         2012         2013         2014  \
0      188042.00    125760.00    849701.00   1543854.00    586894.00   
1      258229.00   1610459.00   3924288.00   2994667.00   1359580.00   
2     2050738.00   2843408.00   1052381.00   4146598.00   3199454.00   
3           0.00         0.00    811074.00   3853785.00   2018957.00   
4     6506574.00   4505678.00  23214194.00  12483621.80   1075159.78   
..           ...          ...          ...          ...          ...   
96          0.00         0.00         0.00         0.00         0.00   
97   10230647.20  10388104.20  28181715.36  29386164.24  46667155.67   
98          0.00         0.00         0.00         0.00         0.00   
99    6328659.24   5594323.10   5629791.20   3900555.00   9764715.00   
100    601039.00    661391.78    433229.00    836228.06   1439233.30   

            2015         2016  
0      718792.00   1009642.00  
1     1615146.00   1698224.00  
2     3975572.40   5109116.70  
3     2900104.40   3550139.70  
4     1975977.40    858602.58  
..           ...          ...  
96          0.00         0.00  
97   36643862.87         0.00  
98          0.00  18248614.10  
99   18681889.00  19170058.00  
100   1788028.80   1908729.31  

[101 rows x 9 columns]

Работа с Архангельской обл-тью

In [7]:
names = ['Архангельская область', 'Архангельская область (кроме Ненецкого автономного округа)', 'Ненецкий автономный округ (Архангельская область)']
data2[(data2['Регион'].isin(names))]

Регион       2009       2010  \
2                               Архангельская область  1781604.1  2050738.0   
3   Архангельская область (кроме Ненецкого автоном...        0.0        0.0   
39  Ненецкий автономный округ (Архангельская область)    81654.9   823265.0   

         2011       2012       2013       2014       2015       2016  
2   2843408.0  1052381.0  4146598.0  3199454.0  3975572.4  5109116.7  
3         0.0   811074.0  3853785.0  2018957.0  2900104.4  3550139.7  
39  1019875.0        0.0   292813.0  1180497.0  1075468.0  1558977.0

In [8]:
for year in ('2009', '2010', '2011'):
    value = (
        data2.loc[data2['Регион'] == names[0], year].values[0]
        - data2.loc[data2['Регион'] == names[2], year].values[0]
    )
    data2.loc[data2['Регион'] == names[1], year] = value
value = (
        data2.loc[data2['Регион'] == names[0], '2012'].values[0]
        - data2.loc[data2['Регион'] == names[1], '2012'].values[0]
    )
data2.loc[data2['Регион'] == names[2], '2012'] = value

In [9]:
data2[(data2['Регион'].isin(names))]

Регион       2009       2010  \
2                               Архангельская область  1781604.1  2050738.0   
3   Архангельская область (кроме Ненецкого автоном...  1699949.2  1227473.0   
39  Ненецкий автономный округ (Архангельская область)    81654.9   823265.0   

         2011       2012       2013       2014       2015       2016  
2   2843408.0  1052381.0  4146598.0  3199454.0  3975572.4  5109116.7  
3   1823533.0   811074.0  3853785.0  2018957.0  2900104.4  3550139.7  
39  1019875.0   241307.0   292813.0  1180497.0  1075468.0  1558977.0

Работа с Тюменской обл-тью

In [10]:
names = ['Тюменская область', 'Тюменская область (кроме Ханты-Мансийского автономного округа-Югры и Ямало-Ненецкого автономного округа)', 'Ханты-Мансийский автономный округ - Югра (Тюменская область)', 'Ямало-Ненецкий автономный округ (Тюменская область)']
data2[data2['Регион'].isin(names)]

Регион         2009  \
84                                  Тюменская область  15970457.79   
85  Тюменская область (кроме Ханты-Мансийского авт...         0.00   
90  Ханты-Мансийский автономный округ - Югра (Тюме...   7077361.03   
99  Ямало-Ненецкий автономный округ (Тюменская обл...   8583498.94   

           2010         2011        2012        2013        2014        2015  \
84  31148351.08  26675182.12  37588368.8  40571677.0  41801013.0  44300522.0   
85         0.00         0.00   3675591.0  12275177.0  15298034.0   4318199.0   
90  24176098.84  19737002.42  26996854.1  24395945.0  16738264.0  21300434.0   
99   6328659.24   5594323.10   5629791.2   3900555.0   9764715.0  18681889.0   

          2016  
84  35173003.0  
85   4545893.0  
90  11457052.0  
99  19170058.0

In [11]:
for year in ('2009', '2010', '2011'):
    value = (
        data2.loc[data2['Регион'] == names[0], year].values[0]
        - data2.loc[data2['Регион'] == names[2], year].values[0]
        - data2.loc[data2['Регион'] == names[3], year].values[0]
    )
    data2.loc[data2['Регион'] == names[1], year] = value

In [12]:
data2[data2['Регион'].isin(names)]

Регион         2009  \
84                                  Тюменская область  15970457.79   
85  Тюменская область (кроме Ханты-Мансийского авт...    309597.82   
90  Ханты-Мансийский автономный округ - Югра (Тюме...   7077361.03   
99  Ямало-Ненецкий автономный округ (Тюменская обл...   8583498.94   

           2010         2011        2012        2013        2014        2015  \
84  31148351.08  26675182.12  37588368.8  40571677.0  41801013.0  44300522.0   
85    643593.00   1343856.60   3675591.0  12275177.0  15298034.0   4318199.0   
90  24176098.84  19737002.42  26996854.1  24395945.0  16738264.0  21300434.0   
99   6328659.24   5594323.10   5629791.2   3900555.0   9764715.0  18681889.0   

          2016  
84  35173003.0  
85   4545893.0  
90  11457052.0  
99  19170058.0

In [13]:
extra_regions = ['Российская Федерация', 'Архангельская область', 'Тюменская область', 'в том числе:', 'Республика Крым', 'г. Севастополь', 'Москва в старых границах', 'Московская обл. в старых границах']
data2 = data2[data2['Регион'].str.endswith('федеральный округ') == False]
data2 = data2[~data2['Регион'].isin(extra_regions)] #~отрицание условия
data2.head()

Регион       2009       2010  \
0                                     Алтайский край   307775.0   188042.0   
1                                   Амурская область  1269189.0   258229.0   
3  Архангельская область (кроме Ненецкого автоном...  1699949.2  1227473.0   
4                               Астраханская область  2117091.0  6506574.0   
5                               Белгородская область   880405.0  1488078.0   

        2011        2012        2013        2014        2015        2016  
0   125760.0    849701.0   1543854.0   586894.00   718792.00  1009642.00  
1  1610459.0   3924288.0   2994667.0  1359580.00  1615146.00  1698224.00  
3  1823533.0    811074.0   3853785.0  2018957.00  2900104.40  3550139.70  
4  4505678.0  23214194.0  12483621.8  1075159.78  1975977.40   858602.58  
5  3335261.8  11797337.3   6062674.3  3454582.70  7141368.36  2425641.00

In [14]:
data2 = data2[data2['Регион'].isin(region_mapping.keys())]
data2.loc[:, 'Регион'] = data2['Регион'].map(region_mapping)
data2.info()

<class 'pandas.core.frame.DataFrame'>
Index: 83 entries, 0 to 100
Data columns (total 9 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2009    83 non-null     float64
 2   2010    83 non-null     float64
 3   2011    83 non-null     float64
 4   2012    83 non-null     float64
 5   2013    83 non-null     float64
 6   2014    83 non-null     float64
 7   2015    83 non-null     float64
 8   2016    83 non-null     float64
dtypes: float64(8), object(1)
memory usage: 6.5+ KB


Инвестиции 2017-2023

In [15]:
data3 = pd.read_excel("data (3).xls", skiprows=2)
data3.drop(0, inplace=True)
data3.rename(columns={'Unnamed: 0':'Регион', 'Unnamed: 1':'Вид оос'}, inplace=True)
data3['Регион'] = data3['Регион'].apply(lambda x: x.strip())
data3.drop('2024', axis=1, inplace=True)
data3 = pd.DataFrame(data3.drop('Вид оос', axis=1).groupby(['Регион']).sum())
data3.reset_index(inplace=True)
data3[data3.drop('Регион', axis=1).columns] = data3[data3.drop('Регион', axis=1).columns].astype(float)
data3.head()

Регион        2017       2018  \
0                                     Алтайский край    177158.6  1499996.0   
1                                   Амурская область   3202654.0  3662197.0   
2                              Архангельская область  13728049.0  3630022.0   
3                     Архангельская область (без АО)         0.0  1373982.0   
4  Архангельская область (кроме Ненецкого автоном...   9077923.0  2531498.0   

        2019        2020        2021       2022       2023  
0   737426.0    693377.0    520729.0   575528.0  2247651.0  
1  2434168.0   7291807.0   7520455.0  3508357.0  2883951.0  
2  2387827.0  17496503.0  18654370.0  3198926.0  3430390.0  
3        0.0         0.0         0.0        0.0        0.0  
4  2075759.0  13846375.0  18755549.0  2898495.0  2286387.0

In [16]:
names = ['Архангельская область', 'Архангельская область (без АО)','Архангельская область (кроме Ненецкого автономного округа)','Ненецкий автономный округ (Архангельская область)',   'Сухопутные территории Архангельской области, входящие в состав Арктической зоны Российской Федерации', 'Сухопутные территории Архангельской области, входящие в состав Арктической зоны РФ']
data3[data3['Регион'].isin(names)]

Регион        2017       2018  \
2                                Архангельская область  13728049.0  3630022.0   
3                       Архангельская область (без АО)         0.0  1373982.0   
4    Архангельская область (кроме Ненецкого автоном...   9077923.0  2531498.0   
61   Ненецкий автономный округ (Архангельская область)   4650126.0  1098524.0   
108  Сухопутные территории Архангельской области, в...   5153817.0        0.0   
109  Сухопутные территории Архангельской области, в...         0.0        0.0   

          2019        2020        2021       2022       2023  
2    2387827.0  17496503.0  18654370.0  3198926.0  3430390.0  
3          0.0         0.0         0.0        0.0        0.0  
4    2075759.0  13846375.0  18755549.0  2898495.0  2286387.0  
61         0.0         0.0         0.0        0.0   515966.0  
108        0.0         0.0         0.0        0.0        0.0  
109        0.0   2777454.0   2283760.0  5338722.0  4441613.0

In [17]:
for year in ('2019', '2020', '2021', '2022'):
    value = (
        data3.loc[data3['Регион'] == names[0], year].values[0]
        - data3.loc[data3['Регион'] == names[2], year].values[0]
    )
    data3.loc[data3['Регион'] == names[3], year] = value

In [18]:
names = ['Архангельская область', 'Ненецкий автономный округ (Архангельская область)', 'Архангельская область (без АО)', 'Архангельская область (кроме Ненецкого автономного округа)', 'Сухопутные территории Архангельской области, входящие в состав Арктической зоны Российской Федерации', 'Сухопутные территории Архангельской области, входящие в состав Арктической зоны РФ']
data3[data3['Регион'].isin(names)]

Регион        2017       2018  \
2                                Архангельская область  13728049.0  3630022.0   
3                       Архангельская область (без АО)         0.0  1373982.0   
4    Архангельская область (кроме Ненецкого автоном...   9077923.0  2531498.0   
61   Ненецкий автономный округ (Архангельская область)   4650126.0  1098524.0   
108  Сухопутные территории Архангельской области, в...   5153817.0        0.0   
109  Сухопутные территории Архангельской области, в...         0.0        0.0   

          2019        2020        2021       2022       2023  
2    2387827.0  17496503.0  18654370.0  3198926.0  3430390.0  
3          0.0         0.0         0.0        0.0        0.0  
4    2075759.0  13846375.0  18755549.0  2898495.0  2286387.0  
61    312068.0   3650128.0   -101179.0   300431.0   515966.0  
108        0.0         0.0         0.0        0.0        0.0  
109        0.0   2777454.0   2283760.0  5338722.0  4441613.0

Из-за того, что в строке Архангельская область за 2021 г. не включены данные  "Обращение с отходами" и "Предприятия по утилизации, обработке (включая обезвреживание) и полигоны захоронения отходов", получилось отрицательное значение. Поэтому прибавим к некорректному значению значения этих данных.

In [19]:
data3.iloc[61, 5] = data3.iloc[61, 5] + 2 * 51777 # данные из той же таблицы

In [20]:
names = ['Тюменская область', 'Тюменская область (без АО)',  'Тюменская область (кроме Ханты-Мансийского автономного округа-Югры и Ямало-Ненецкого автономного округа)','Ханты-Мансийский автономный округ - Югра (Тюменская область)','Ямало-Ненецкий автономный округ (Тюменская область)']
data3[data3['Регион'].isin(names)]

Регион        2017  \
123                                  Тюменская область  71866651.0   
124                         Тюменская область (без АО)         0.0   
125  Тюменская область (кроме Ханты-Мансийского авт...   7525544.0   
130  Ханты-Мансийский автономный округ - Югра (Тюме...  11935067.0   
137  Ямало-Ненецкий автономный округ (Тюменская обл...  52406040.0   

           2018        2019        2020         2021         2022         2023  
123  34514846.0  48166298.0  36610918.0  200703377.0  110218013.0  134896085.0  
124   1557120.0         0.0         0.0          0.0          0.0          0.0  
125   3797546.0   3340385.0  15935233.0    2976761.0    2462814.0    2579635.0  
130   7787591.0  13565306.0   4527197.0    4745699.0   11666952.0   21813424.0  
137  22929709.0  29618308.0  15303209.0  191201596.0   94477915.0  105498709.0

In [21]:
extra_regions = ['Российская Федерация', 'Архангельская область', 'Тюменская область', 'в том числе:', 'Республика Крым', 'г. Севастополь', 'Москва в старых границах', 'Московская обл. в старых границах', 'Архангельская область (без АО)',  'Сухопутные территории Архангельской области, входящие в состав Арктической зоны Российской Федерации', 'Сухопутные территории Архангельской области, входящие в состав Арктической зоны РФ', 'Места традиционного проживания и традиционной хозяйственной деятельности коренных малочисленных народов Севера, Сибири и Дальнего Востока на территории Тюменской области', 'Тюменская область (без АО)']
data3 = data3[data3['Регион'].str.endswith('федеральный округ') == False]
data3 = data3[~data3['Регион'].isin(extra_regions)] #~отрицание условия
data3 = data3[data3['Регион'].isin(region_mapping.keys())]
data3.loc[:, 'Регион'] = data3['Регион'].map(region_mapping)

Данные по Республике Ингушетии отсутствуют в данных ЕМИСС, поэтому взяты из официального Постановления Правительства Республики Ингушетия от 28.05.2020 № 72 «О внесении изменений в государственную программу Республики Ингушетия «Охрана и защита окружающей среды»».​

В документе в разделе "Объемы бюджетных ассигнований государственной программы" указаны объемы финансирования по годам, включая разбивку по источникам (федеральный бюджет, республиканский бюджет, внебюджетные источники).

Конкретно цифры находятся в паспорте Программы, который был изменен этим постановлением.

In [22]:
data3.loc[139] = ["Республика Ингушетия", 252000, 69900, 751900, 7363100, 210200, 113700, 1219900]

In [23]:
data3.tail()

Регион         2017        2018        2019  \
134   Чувашская Республика — Чувашия    961636.47   1196739.0    769833.0   
135       Чукотский автономный округ   1646837.00    543841.0    651448.0   
137  Ямало-Ненецкий автономный округ  52406040.00  22929709.0  29618308.0   
138              Ярославская область   1825997.21   2372085.0   4670890.0   
139             Республика Ингушетия    252000.00     69900.0    751900.0   

           2020         2021        2022         2023  
134   2767669.0    1939780.0   1177279.0     314596.0  
135    710223.0      89950.0      9990.0     800015.0  
137  15303209.0  191201596.0  94477915.0  105498709.0  
138   4537756.0    4521503.0   5632013.0    9917079.0  
139   7363100.0     210200.0    113700.0    1219900.0

In [24]:
all_invest = pd.merge(data1, data2, how='outer', on='Регион')
all_invest = pd.merge(all_invest, data3, how='outer', on='Регион')

In [25]:
all_invest.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 83 entries, 0 to 82
Data columns (total 25 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2000    83 non-null     float64
 2   2001    83 non-null     float64
 3   2002    83 non-null     float64
 4   2003    83 non-null     float64
 5   2004    83 non-null     float64
 6   2005    83 non-null     float64
 7   2006    83 non-null     float64
 8   2007    83 non-null     float64
 9   2008    83 non-null     float64
 10  2009    83 non-null     float64
 11  2010    83 non-null     float64
 12  2011    83 non-null     float64
 13  2012    83 non-null     float64
 14  2013    83 non-null     float64
 15  2014    83 non-null     float64
 16  2015    83 non-null     float64
 17  2016    83 non-null     float64
 18  2017    83 non-null     float64
 19  2018    83 non-null     float64
 20  2019    83 non-null     float64
 21  2020    83 non-null     float64
 22  2021

## Врп

In [26]:
vrp = pd.read_excel('Pril-Region_Pokaz_2024/Раздел 8 - Валовой региональный продукт.xlsx', sheet_name=1, skiprows=8)
vrp.rename(columns={'Unnamed: 0':'Регион'}, inplace=True)
vrp['Регион'] = vrp['Регион'].apply(lambda x: x.strip())
vrp.drop(['Unnamed: 24', 'Unnamed: 25'], axis=1, inplace=True)
vrp[vrp.drop('Регион', axis=1).columns] = vrp[vrp.drop('Регион', axis=1).columns].astype(float)
vrp.head()

Регион       2000       2001  \
0  Валовой региональный продукт по субъектам Росс...  5753671.6  7170968.2   
1                      Центральный федеральный округ  1841498.9  2243525.0   
2                               Белгородская область    42074.5    49941.8   
3                                   Брянская область    24650.5    30110.3   
4                               Владимирская область    33017.7    42075.4   

        2002        2003        2004        2005        2006        2007  \
0  8741219.2  10742423.3  13964305.4  18034385.2  22492119.6  27963955.6   
1  2878664.5   3577142.5   4617086.1   6278359.2   7965169.5  10208917.7   
2    62404.4     76054.5    114409.3    144987.8    178846.1    237013.3   
3    37374.1     43700.3     51003.4     66692.3     82100.4    102706.2   
4    50359.9     61818.6     74207.0     86926.8    112841.7    146663.0   

         2008  ...        2013        2014        2015        2016  \
0  33908756.7  ...  54103000.3  59188270.3  65750633.6  74120174.8   
1  12674395.4  ...  19160905.7  20866362.0  22663758.1  25995587.5   
2    317656.3  ...    569006.4    619677.7    693379.4    778027.8   
3    125834.4  ...    219502.8    242722.4    271782.5    316489.4   
4    175395.7  ...    306641.4    328064.2    368489.2    431549.8   

         2017        2018        2019        2020          2021         2022  
0  79745093.9  90202901.5  95060662.3  94410215.3  1.221997e+08  140670816.5  
1  27915455.0  31191756.5  33139758.1  34167817.7  4.159234e+07   47367524.5  
2    837306.8    911597.9    955329.2    997330.9  1.359966e+06    1311232.6  
3    341177.8    367157.1    399113.8    414179.4  4.826694e+05     549347.1  
4    449849.2    480027.8    535493.4    553092.4  7.303288e+05     780490.4  

[5 rows x 24 columns]

In [27]:
names = ['Архангельская область','Ненецкий автономный округ', 'Архангельская область без автономного округа']
vrp[vrp['Регион'].isin(names)]

Регион     2000     2001     2002  \
23                         Архангельская область  61806.9  67274.7  83158.8   
25                     Ненецкий автономный округ  11924.0  11883.5  16480.8   
26  Архангельская область без автономного округа      NaN      NaN      NaN   

        2003      2004      2005      2006      2007      2008  ...      2013  \
23  103951.3  142564.7  166433.4  215932.7  268672.1  289755.9  ...  500095.1   
25   25040.6   39586.7   44718.3   67248.4   97838.3   91476.4  ...  173170.2   
26       NaN       NaN       NaN       NaN       NaN       NaN  ...  326924.9   

        2014      2015      2016      2017      2018      2019      2020  \
23  542695.3  627698.1  713043.9  759206.5  865723.3  889974.4  781819.0   
25  187009.8  227193.5  261772.5  265790.9  320405.7  330999.0  230674.3   
26  355685.5  400504.6  451271.4  493415.6  545317.6  558975.4  551144.7   

         2021       2022  
23  1080494.9  1188187.1  
25   409388.3   488014.4  
26   671106.7   700172.7  

[3 rows x 24 columns]

In [28]:
for year in range(2000, 2011):
    value = (
        vrp.loc[vrp['Регион'] == names[0], year].values[0]
        - vrp.loc[vrp['Регион'] == names[1], year].values[0]
    )
    vrp.loc[vrp['Регион'] == names[2], year] = value

In [29]:
names = ['Архангельская область','Ненецкий автономный округ', 'Архангельская область без автономного округа']
vrp[vrp['Регион'].isin(names)]

Регион     2000     2001     2002  \
23                         Архангельская область  61806.9  67274.7  83158.8   
25                     Ненецкий автономный округ  11924.0  11883.5  16480.8   
26  Архангельская область без автономного округа  49882.9  55391.2  66678.0   

        2003      2004      2005      2006      2007      2008  ...      2013  \
23  103951.3  142564.7  166433.4  215932.7  268672.1  289755.9  ...  500095.1   
25   25040.6   39586.7   44718.3   67248.4   97838.3   91476.4  ...  173170.2   
26   78910.7  102978.0  121715.1  148684.3  170833.8  198279.5  ...  326924.9   

        2014      2015      2016      2017      2018      2019      2020  \
23  542695.3  627698.1  713043.9  759206.5  865723.3  889974.4  781819.0   
25  187009.8  227193.5  261772.5  265790.9  320405.7  330999.0  230674.3   
26  355685.5  400504.6  451271.4  493415.6  545317.6  558975.4  551144.7   

         2021       2022  
23  1080494.9  1188187.1  
25   409388.3   488014.4  
26   671106.7   700172.7  

[3 rows x 24 columns]

In [30]:
names = ['Тюменская область','Ханты-Мансийский автономный округ – Югра', 'Ямало-Ненецкий автономный округ', 'Тюменская область без автономных округов']
vrp[vrp['Регион'].isin(names)]

Регион      2000      2001      2002  \
69                         Тюменская область  570790.2  753119.2  898722.4   
71  Ханты-Мансийский автономный округ – Югра  403822.2  497981.4  552483.5   
72           Ямало-Ненецкий автономный округ  117100.8  184315.9  262447.4   
73  Тюменская область без автономных округов       NaN       NaN       NaN   

         2003       2004       2005       2006       2007       2008  ...  \
69  1117514.4  1536733.7  2215584.4  2551355.4  2758813.1  3121401.3  ...   
71   717219.7   956196.5  1399335.9  1594097.1  1728340.2  1937159.1  ...   
72   283181.2   355718.4   441721.8   546365.8   594678.6   719397.0  ...   
73        NaN        NaN        NaN        NaN        NaN        NaN  ...   

         2013       2014       2015       2016       2017       2018  \
69  4950207.4  5295348.5  5851557.8  6125961.8  7097070.2  8875003.7   
71  2729122.4  2860498.9  3154058.7  3130196.4  3557367.1  4506739.7   
72  1375878.8  1633382.2  1791825.6  2028234.6  2456293.7  3051613.1   
73   845206.2   801467.4   905673.5   967530.8  1083409.4  1316650.9   

         2019       2020        2021        2022  
69  8952460.5  7280595.6  11492922.3  13964548.8  
71  4558879.8  3341832.0   5700228.3   6894316.5  
72  3158827.6  2767734.5   4219244.6   5350247.9  
73  1234753.1  1171029.2   1573449.4   1719984.4  

[4 rows x 24 columns]

In [31]:
for year in range(2000, 2011):
    value = (
        vrp.loc[vrp['Регион'] == names[0], year].values[0]
        - vrp.loc[vrp['Регион'] == names[1], year].values[0]
        - vrp.loc[vrp['Регион'] == names[2], year].values[0]
    )
    vrp.loc[vrp['Регион'] == names[3], year] = value

In [32]:
names = ['Тюменская область','Ханты-Мансийский автономный округ – Югра', 'Ямало-Ненецкий автономный округ', 'Тюменская область без автономных округов']
vrp[vrp['Регион'].isin(names)]

Регион      2000      2001      2002  \
69                         Тюменская область  570790.2  753119.2  898722.4   
71  Ханты-Мансийский автономный округ – Югра  403822.2  497981.4  552483.5   
72           Ямало-Ненецкий автономный округ  117100.8  184315.9  262447.4   
73  Тюменская область без автономных округов   49867.2   70821.9   83791.5   

         2003       2004       2005       2006       2007       2008  ...  \
69  1117514.4  1536733.7  2215584.4  2551355.4  2758813.1  3121401.3  ...   
71   717219.7   956196.5  1399335.9  1594097.1  1728340.2  1937159.1  ...   
72   283181.2   355718.4   441721.8   546365.8   594678.6   719397.0  ...   
73   117113.5   224818.8   374526.7   410892.5   435794.3   464845.2  ...   

         2013       2014       2015       2016       2017       2018  \
69  4950207.4  5295348.5  5851557.8  6125961.8  7097070.2  8875003.7   
71  2729122.4  2860498.9  3154058.7  3130196.4  3557367.1  4506739.7   
72  1375878.8  1633382.2  1791825.6  2028234.6  2456293.7  3051613.1   
73   845206.2   801467.4   905673.5   967530.8  1083409.4  1316650.9   

         2019       2020        2021        2022  
69  8952460.5  7280595.6  11492922.3  13964548.8  
71  4558879.8  3341832.0   5700228.3   6894316.5  
72  3158827.6  2767734.5   4219244.6   5350247.9  
73  1234753.1  1171029.2   1573449.4   1719984.4  

[4 rows x 24 columns]

In [33]:
extra_regions = ['Российская Федерация', 'Архангельская область', 'Тюменская область', 'в том числе:', 'Республика Крым', 'г. Севастополь']
vrp = vrp[vrp['Регион'].str.endswith('федеральный округ') == False]
vrp = vrp[~vrp['Регион'].isin(extra_regions)] #~отрицание условия
vrp

Регион       2000       2001  \
0   Валовой региональный продукт по субъектам Росс...  5753671.6  7170968.2   
2                                Белгородская область    42074.5    49941.8   
3                                    Брянская область    24650.5    30110.3   
4                                Владимирская область    33017.7    42075.4   
5                                 Воронежская область    49523.9    60014.6   
..                                                ...        ...        ...   
93                                   Амурская область    26315.2    39052.8   
94                                Магаданская область    13009.5    17151.6   
95                                Сахалинская область    34777.0    47140.1   
96                       Еврейская автономная область     3784.0     4788.9   
97                         Чукотский автономный округ     3931.4     6944.6   

         2002        2003        2004        2005        2006        2007  \
0   8741219.2  10742423.3  13964305.4  18034385.2  22492119.6  27963955.6   
2     62404.4     76054.5    114409.3    144987.8    178846.1    237013.3   
3     37374.1     43700.3     51003.4     66692.3     82100.4    102706.2   
4     50359.9     61818.6     74207.0     86926.8    112841.7    146663.0   
5     83001.1    100143.3    117197.6    133586.6    166176.5    222811.9   
..        ...         ...         ...         ...         ...         ...   
93    45717.5     53199.9     64250.2     76861.2     95090.9    111761.2   
94    22374.8     24325.9     24612.3     27167.8     31203.2     35314.4   
95    47139.8     63139.2     91729.6    121014.1    166105.4    286273.0   
96     6838.5      8564.6     11230.9     14204.2     17976.8     23726.1   
97    10155.4     13501.1     12357.8     12355.4     15538.0     20984.1   

          2008  ...        2013        2014        2015        2016  \
0   33908756.7  ...  54103000.3  59188270.3  65750633.6  74120174.8   
2     317656.3  ...    569006.4    619677.7    693379.4    778027.8   
3     125834.4  ...    219502.8    242722.4    271782.5    316489.4   
4     175395.7  ...    306641.4    328064.2    368489.2    431549.8   
5     287072.1  ...    611720.4    717667.2    805969.6    827928.6   
..         ...  ...         ...         ...         ...         ...   
93    131563.7  ...    210700.9    232053.0    277380.4    297531.0   
94     42053.8  ...     88905.9     96936.8    125798.3    153879.8   
95    333581.6  ...    671743.6    799165.4    837495.2    762510.3   
96     23977.0  ...     38428.7     41948.1     44554.8     48563.2   
97     30558.7  ...     44466.9     57751.3     61735.5     72174.1   

          2017        2018        2019        2020          2021         2022  
0   79745093.9  90202901.5  95060662.3  94410215.3  1.221997e+08  140670816.5  
2     837306.8    911597.9    955329.2    997330.9  1.359966e+06    1311232.6  
3     341177.8    367157.1    399113.8    414179.4  4.826694e+05     549347.1  
4     449849.2    480027.8    535493.4    553092.4  7.303288e+05     780490.4  
5     873429.4    951292.3   1001790.3   1062765.4  1.273852e+06    1377736.5  
..         ...         ...         ...         ...           ...          ...  
93    299181.0    334164.4    395617.2    449317.5  5.482698e+05     603837.3  
94    161851.2    176370.6    214414.9    285146.0  3.201596e+05     315919.1  
95    784503.4   1233164.7   1172226.1   1001689.4  1.237950e+06    1530380.6  
96     55268.0     54577.8     56847.6     63177.1  7.915690e+04      80676.2  
97     72873.8     83422.6     94995.0    119949.4  1.400263e+05     141042.1  

[84 rows x 24 columns]

In [34]:
vrp = vrp[vrp['Регион'].isin(region_mapping.keys())]
vrp.loc[:, 'Регион'] = vrp['Регион'].map(region_mapping)

In [35]:
vrp[vrp[2001].isna()]


Регион  2000  2001  2002  2003  2004     2005     2006  \
49  Чеченская Республика   NaN   NaN   NaN   NaN   NaN  22898.9  32344.4   

       2007     2008  ...      2013      2014      2015      2016      2017  \
49  48056.1  66273.8  ...  122402.8  148942.1  154401.4  201634.8  215975.9   

        2018      2019      2020      2021      2022  
49  223853.7  241643.4  251873.2  278784.6  315069.6  

[1 rows x 24 columns]

В Чеченской Республике с 26 августа 1999 года по 16 апреля 2009 года шла Вторая чеченская война, поэтому корректных данных с 2000 по 2023 год нет, удадляем эту строку

In [36]:
vrp = vrp[vrp['Регион'] != "Чеченская Республика"]

In [37]:
vrp.info()

<class 'pandas.core.frame.DataFrame'>
Index: 82 entries, 2 to 97
Data columns (total 24 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  82 non-null     object 
 1   2000    82 non-null     float64
 2   2001    82 non-null     float64
 3   2002    82 non-null     float64
 4   2003    82 non-null     float64
 5   2004    82 non-null     float64
 6   2005    82 non-null     float64
 7   2006    82 non-null     float64
 8   2007    82 non-null     float64
 9   2008    82 non-null     float64
 10  2009    82 non-null     float64
 11  2010    82 non-null     float64
 12  2011    82 non-null     float64
 13  2012    82 non-null     float64
 14  2013    82 non-null     float64
 15  2014    82 non-null     float64
 16  2015    82 non-null     float64
 17  2016    82 non-null     float64
 18  2017    82 non-null     float64
 19  2018    82 non-null     float64
 20  2019    82 non-null     float64
 21  2020    82 non-null     float64
 22  2021    8

In [38]:
vrp_2023 = pd.read_excel("data (4).xls", skiprows=2)
vrp_2023.drop(0, inplace=True)
vrp_2023.drop('Unnamed: 1', axis=1, inplace=True)
vrp_2023.rename(columns={'Unnamed: 0':'Регион'}, inplace=True)
vrp_2023['Регион'] = vrp_2023['Регион'].apply(lambda x: x.strip())
vrp_2023[vrp_2023.drop('Регион', axis=1).columns] = vrp_2023[vrp_2023.drop('Регион', axis=1).columns].astype(float)
extra_regions = ['Российская Федерация', 'Архангельская область', 'Тюменская область', 'в том числе:', 'Республика Крым', 'г. Севастополь']
vrp_2023 = vrp_2023[vrp_2023['Регион'].str.endswith('федеральный округ') == False]
vrp_2023 = vrp_2023[~vrp_2023['Регион'].isin(extra_regions)] #~отрицание условия
vrp_2023 = vrp_2023[vrp_2023['Регион'].isin(region_mapping.keys())]
vrp_2023.loc[:, 'Регион'] = vrp_2023['Регион'].map(region_mapping)
vrp_2023.info()

<class 'pandas.core.frame.DataFrame'>
Index: 83 entries, 3 to 96
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2023    83 non-null     float64
dtypes: float64(1), object(1)
memory usage: 1.9+ KB


In [39]:
all_vrp = pd.merge(vrp, vrp_2023, how='outer')

In [40]:
all_vrp.head()

Регион     2000     2001     2002     2003      2004  \
0         Алтайский край  46736.8  61854.4  73107.4  88733.3  114840.5   
1       Амурская область  26315.2  39052.8  45717.5  53199.9   64250.2   
2  Архангельская область  49882.9  55391.2  66678.0  78910.7  102978.0   
3   Астраханская область  28115.7  32274.3  40994.9  50659.8   56710.9   
4   Белгородская область  42074.5  49941.8  62404.4  76054.5  114409.3   

       2005      2006      2007      2008  ...      2014      2015      2016  \
0  135686.4  173810.5  223563.4  259343.1  ...  446023.8  487903.3  532401.5   
1   76861.2   95090.9  111761.2  131563.7  ...  232053.0  277380.4  297531.0   
2  121715.1  148684.3  170833.8  198279.5  ...  355685.5  400504.6  451271.4   
3   70127.6   85112.1  100359.2  147549.1  ...  296319.3  322303.0  368485.4   
4  144987.8  178846.1  237013.3  317656.3  ...  619677.7  693379.4  778027.8   

       2017      2018      2019      2020       2021       2022          2023  
0  545303.0  579740.5  628146.1  664129.2   869672.1   944894.1  1.024355e+09  
1  299181.0  334164.4  395617.2  449317.5   548269.8   603837.3  7.938519e+08  
2  493415.6  545317.6  558975.4  551144.7   671106.7   700172.7  7.615896e+08  
3  442608.8  579210.0  601811.2  527290.2   663770.6   765239.3  7.777183e+08  
4  837306.8  911597.9  955329.2  997330.9  1359965.6  1311232.6  1.341409e+09  

[5 rows x 25 columns]

In [41]:
all_vrp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 83 entries, 0 to 82
Data columns (total 25 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  83 non-null     object 
 1   2000    82 non-null     float64
 2   2001    82 non-null     float64
 3   2002    82 non-null     float64
 4   2003    82 non-null     float64
 5   2004    82 non-null     float64
 6   2005    82 non-null     float64
 7   2006    82 non-null     float64
 8   2007    82 non-null     float64
 9   2008    82 non-null     float64
 10  2009    82 non-null     float64
 11  2010    82 non-null     float64
 12  2011    82 non-null     float64
 13  2012    82 non-null     float64
 14  2013    82 non-null     float64
 15  2014    82 non-null     float64
 16  2015    82 non-null     float64
 17  2016    82 non-null     float64
 18  2017    82 non-null     float64
 19  2018    82 non-null     float64
 20  2019    82 non-null     float64
 21  2020    82 non-null     float64
 22  2021

# Объединение датасета

In [42]:
df = df.melt(id_vars=['Регион'], var_name='Год', value_name='Объем_сточных_вод_млн_м3')
df = df[['Регион', 'Год', 'Объем_сточных_вод_млн_м3']]
df['Год'] = df['Год'].astype(int)

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 3 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Регион                    1992 non-null   object 
 1   Год                       1992 non-null   int32  
 2   Объем_сточных_вод_млн_м3  1992 non-null   float64
dtypes: float64(1), int32(1), object(1)
memory usage: 39.0+ KB


In [44]:
all_invest = all_invest.melt(id_vars=['Регион'], var_name='Год', value_name='Инвестиции_в_ООС')
all_invest = all_invest[['Регион', 'Год', 'Инвестиции_в_ООС']]
all_invest['Год'] = all_invest['Год'].astype(int)

In [45]:
all_invest.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Регион            1992 non-null   object 
 1   Год               1992 non-null   int32  
 2   Инвестиции_в_ООС  1992 non-null   float64
dtypes: float64(1), int32(1), object(1)
memory usage: 39.0+ KB


In [46]:
all_vrp = all_vrp.melt(id_vars=['Регион'], var_name='Год', value_name='ВРП')
all_vrp = all_vrp[['Регион', 'Год', 'ВРП']]
all_vrp['Год'] = all_vrp ['Год'].astype(int)


In [47]:
all_vrp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Регион  1992 non-null   object 
 1   Год     1992 non-null   int32  
 2   ВРП     1969 non-null   float64
dtypes: float64(1), int32(1), object(1)
memory usage: 39.0+ KB


In [48]:
all_df = pd.merge(pd.merge(df, all_invest, how='outer', on=['Регион', 'Год']), all_vrp, how='outer', on=['Регион', 'Год'])

In [49]:
all_df

Регион   Год  Объем_сточных_вод_млн_м3  Инвестиции_в_ООС  \
0         Алтайский край  2000                     31.00          176660.0   
1         Алтайский край  2001                     34.00          111590.0   
2         Алтайский край  2002                     36.00           67466.0   
3         Алтайский край  2003                     36.00           50114.0   
4         Алтайский край  2004                     36.00           38270.0   
...                  ...   ...                       ...               ...   
1987  г. Санкт-Петербург  2019                    920.33        13004495.0   
1988  г. Санкт-Петербург  2020                    914.37        18796854.0   
1989  г. Санкт-Петербург  2021                    945.06        21438182.0   
1990  г. Санкт-Петербург  2022                    932.70        11413665.0   
1991  г. Санкт-Петербург  2023                    936.00        29053267.0   

               ВРП  
0     4.673680e+04  
1     6.185440e+04  
2     7.310740e+04  
3     8.873330e+04  
4     1.148405e+05  
...            ...  
1987  5.186129e+06  
1988  5.332558e+06  
1989  9.420445e+06  
1990  1.116644e+07  
1991  1.090803e+10  

[1992 rows x 5 columns]

In [50]:
all_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1992 entries, 0 to 1991
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Регион                    1992 non-null   object 
 1   Год                       1992 non-null   int32  
 2   Объем_сточных_вод_млн_м3  1992 non-null   float64
 3   Инвестиции_в_ООС          1992 non-null   float64
 4   ВРП                       1969 non-null   float64
dtypes: float64(3), int32(1), object(1)
memory usage: 70.2+ KB


In [51]:
all_df.to_excel('dataset.xlsx')
all_df.to_csv('dataset.csv')

# Дополнительные столбцы

**Территориальная принадлежность**
 - Округ - из столбцов Регион
 
**Использование воды**
 - 7.5. ИСПОЛЬЗОВАНИЕ СВЕЖЕЙ ВОДЫ (2000-2023)
 - 7.6. ОБЪЕМ ОБОРОТНОЙ И ПОСЛЕДОВАТЕЛЬНО ИСПОЛЬЗУЕМОЙ ВОДЫ (2000-2023)
 
**Фонды**
 - 9.3 СТЕПЕНЬ ИЗНОСА ОСНОВНЫХ ФОНДОВ (2001 - 2023)
 
**Загрязнение**
 - 7.7. СБРОС ЗАГРЯЗНЕННЫХ СТОЧНЫХ ВОД В ПОВЕРХНОСТНЫЕ ВОДНЫЕ ОБЪЕКТЫ
 
**Население**
 - Численность населения (2000-2023)
 - 1.11.СМЕРНОСТЬ НАСЕЛЕНИЯ В ТРУДОСПОСОБНОМ ВОЗРАСТЕ (2000-2023)

 - 1.10 Общие коэффициенты смертности (2000-2023) 

 - 1.12 Смертность населения (без показателя смертности от внешних причин) (2011-2023)
 
**Инвестиции**
 - Инвестиции на охрану окр природы (разные источники)
 - 10.1 ИНВЕСТИЦИИ В ОСНОВНОЙ КАПИТАЛ (2000-2023)
 
**Охрана окружающей природы**
 - Количество очистительных сооружений (из разных источников)  (https://fedstat.ru/indicator/34033) (2000-2024)
 
 
**Туризм**
 - 6.9.1. Численность российских туристов, отправленных туристскими фирмами, в туры по России (2002-2023)
 
**Здравоохранение**
 - 5.9.1 ЗАБОЛЕВАЕМОСТЬ на 1000 человек населения ПО ОСНОВНЫМ КЛАССАМ БОЛЕЗНЕЙ Некоторые инфекционные и паразитарные болезни(2004-2023)
 - 5.9.15. Травмы, отравления и некоторые другие последствия воздействия внешних причин (2004-2023)
 
**Сельское, лесное хозяйство, рыболовство и рыбоводство**
 - 13.41. УЛОВ РЫБЫ И ДОБЫЧА ДРУГИХ ВОДНЫХ БИОРЕСУРСОВ (2016-2023)
 - 13.40.ОСНОВНЫЕ ПОКАЗАТЕЛИ ПО ВИДУ ЭКОНОМИЧЕСКОЙ ДЕЯТЕЛЬНОСТИ «РЫБОВОДСТВО»)(2005-2023)
 
**Промышленность**
 - 12.2.7  Водоснабжение; водоотведение, организация сбора и утилизации отходов, деятельность по ликвидации загрязнений» в соответствии с ОКВЭД21 (2017-2023)

 - 12.8. ИНДЕКСЫ ПРОИЗВОДСТВА по виду экономической деятельности «Водоснабжение; водоотведение, организация сбора и утилизации отходов, деятельность по ликвидации загрязнений»  в соответствии с ОКВЭД21) (2015-2023)
 
**Коммунальные услуги**
 - 15.14. ОБЪЕМ КОММУНАЛЬНЫХ УСЛУГ НАСЕЛЕНИЮ (2000-2023)
 
**Цены и тарифы**
 - 20.11.9. Водоснабжение; водоотведение, организация сбора и утилизации отходов, деятельность по ликвидации загрязнений (2016-2023)
